plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -3190,16 +3146,8 @@ const std::set<account_name>& producer_plugin::producer_accounts() const {
    return my->_producers;
 }
 
-void producer_plugin::register_update_produced_block_metrics(std::function<void(producer_plugin::produced_block_metrics)>&& fun) {
-   my->_update_produced_block_metrics = std::move(fun);
-}
-
 void producer_plugin::register_update_speculative_block_metrics(std::function<void(speculative_block_metrics)> && fun) {
```

Comment:
I think should take a value, not a rvalue.

Reply:
why?

Reply:
Why accept only rvalues? On this case it works because if you pass a lambda, and intermediate `std::function` is created so it is always a rvalue.

I think we should only require `rvalues` when we have a specific reason to not want any copying.